src/registrar/models/models.py

```diff
@@ -83,3 +87,179 @@ def __str__(self):
                 return self.user.username
             except ObjectDoesNotExist:
                 return "No username"
+
+
+class Website(models.Model):
+
+    """Keep domain names in their own table so that applications can refer to
+    many of them."""
+
+    # domain names have strictly limited lengths, 255 characters is more than
+    # enough.
+    website = models.CharField(max_length=255, null=False, help_text="")
+
+    # a domain name is alphanumeric or hyphen, up to 63 characters, doesn't
+    # begin or end with a hyphen, followed by a TLD of 2-6 alphabetic characters
+    DOMAIN_REGEX = re.compile(r"^(?!-)[A-Za-z0-9-]{1,63}(?<!-)\.[A-Za-z]{2,6}")
+
+    @classmethod
+    def string_could_be_domain(cls, domain):
+        """Return True if the string could be a domain name, otherwise False.
+
+        TODO: when we have a Domain class, this could be a classmethod there.
+        """
+        if cls.DOMAIN_REGEX.match(domain):
+            return True
+        return False
+
+
+class Contact(models.Model):
+
+    """Contact information follows a similar pattern for each contact."""
+
+    first_name = models.TextField(null=True, help_text="First name")
+    middle_name = models.TextField(null=True, help_text="Middle name")
+    last_name = models.TextField(null=True, help_text="Last name")
+    title = models.TextField(null=True, help_text="Title")
+    email = models.TextField(null=True, help_text="Email")
+    phone = models.TextField(null=True, help_text="Phone")
+
+
+class DomainApplication(TimeStampedModel):
+
+    STARTED = "started"
+    SUBMITTED = "submitted"
+    INVESTIGATING = "investigating"
+    APPROVED = "approved"
+    STATUS_CHOICES = [
+        (STARTED, STARTED),
+        (SUBMITTED, SUBMITTED),
+        (INVESTIGATING, INVESTIGATING),
+        (APPROVED, APPROVED),
+    ]
+    status = FSMField(
+        choices=STATUS_CHOICES,  # possible states as an array of constants
+        default=STARTED,  # sensible default
+        protected=True,  # cannot change state directly, must use methods!
```

Comment:
I've been using this branch for my audit logging work and I've noticed that you cannot make changes within the Django admin with this set to True.

I'm not sure what to advise. I think we can defer any work on the admin for right now, but perhaps setting this to `False` would make future dev work a little easier? OTOH, I like that it is catching instances of changing state without using the proper methods.